metadata/src/main/java/org/apache/kafka/controller/errors/ControllerExceptions.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.controller.errors;
+
+import org.apache.kafka.common.errors.TimeoutException;
+
+import java.util.concurrent.ExecutionException;
+
+
+public class ControllerExceptions {
+    /**
+     * Check if an exception is a normal timeout exception.
+     *
+     * @param exception     The exception to check.
+     * @return              True if the exception is a normal timeout exception and NOT a timeout
+     *                      exception generated by the controller shutting down.
+     */
+    public static boolean isNormalTimeoutException(Throwable exception) {
+        if (exception == null) return false;
+        if (exception instanceof ExecutionException) {
+            exception = exception.getCause();
+            if (exception == null) return false;
+        }
+        if (!(exception instanceof TimeoutException)) return false;
+        if (exception.getMessage() != null &&
+                exception.getMessage().equals("The controller is shutting down.")) {
+            return false;
+        }
+        return true;
```

Comment:
This seems brittle.  Is there a way we can identify the specific thing we are interested in?

Reply:
I removed the check for the exception text. This is something I'll revisit in a follow-on PR (there are a few corner cases around error handling and queue shutdown to fix up)